### PR TITLE
[UI/Eslint] Enabled important eslint checks + removed all the errors 

### LINKED ISSUE #5953: [UI/Eslint] Enable important eslint checks
#### Current Behavior

The eslint is a great complement with JS to catch the errors in front during parse time. But some important checks being on `off` state introduces bugs in the code and failures that can be otherwise rectified in advance. Such checks are jsx keys and "no-dupe-keys". 

#### Desired Behavior
```
 "react/jsx-key" : "off",
 "no-dupe-keys" : "off",
```

WARN on "react/jsx-key"
Error on "no-dupe-keys"

#### Environment

- Browser: Chrome Safari Firefox
- Host OS: Mac Linux Windows
- Meshery Server Version: stable-v
- Meshery Client Version: stable-v
- Platform: Docker Kubernetes

---

#### Contributor [Guides](https://docs.meshery.io/project/contributing) and [Handbook](https://layer5.io/community/handbook)

- 🎨 Wireframes and designs for Meshery UI in [Figma](https://www.figma.com/file/SMP3zxOjZztdOLtgN4dS2W/Meshery-UI)
- 🖥 [Contributing to Meshery UI](https://docs.meshery.io/project/contributing/contributing-ui)
- 🙋🏾🙋🏼 Questions: [Discussion Forum](https://discuss.layer5.io) and [Community Slack](http://slack.layer5.io)

### PR DESCRIPTION
**Pull request by kamalbuilds**
**Description**
[UI/Eslint] Enabled important eslint checks + removed all the errors 

This PR fixes #5953

**Notes for Reviewers**

I took the reference from this offical doc to remove the corresponding errors while applying eslint checks 
https://developer.mozilla.org/en-US/docs/Web/API/Request/credentials

**[Signed commits](https://github.com/meshery/meshery/blob/master/CONTRIBUTING.md#signing-off-on-commits-developer-certificate-of-origin)**
- [x] Yes, I signed my commits.

### CODE REVIEW

**Comment by Abhishek-kumar09** [comment]
There are one or two errors that needs to be resolved: https://github.com/meshery/meshery/runs/8106013611?check_suite_focus=true

File: ui/.eslintrc.js [hunk 1]
```diff
@@ -64,16 +60,15 @@ module.exports = {
       }
     ],
     "react/react-in-jsx-scope" : "off",
-    "no-undef" : "error",
-    "react/prop-types" : 0,
+    "no-undef" : "off",
```
Comment (Abhishek-kumar09) [inline ui/.eslintrc.js:63]: `"no-undef": "off"` 😖

**Review by Abhishek-kumar09 — CHANGES_REQUESTED** [review]
Oh, good @legendarykamal Big changes, thanks for pushing on it. the undef is set to "off" state mistakenly, please revert.

File: ui/.eslintrc.js [hunk 2]
```diff
@@ -39,10 +39,6 @@ module.exports = {
     "keyword-spacing" : "error",
     "no-trailing-spaces" : "error",
     "object-curly-spacing" : ["error", "always"],
-    "arrow-spacing" : ["error", {
-      "before" : true,
-      "after" : true,
-    }],
```
Comment (Nikhil-Ladha) [inline ui/.eslintrc.js:45]: This is a "good-to-have" check I think. Keeps code clean, and well oriented.

Comment (legendarykamal) [reply ui/.eslintrc.js:45]: dont worry @Nikhil-Ladha I hadn't removed it , It was declared same two times , so I removed at one place :smiley:

File: ui/.eslintrc.js (hunk 1 above)
Comment (legendarykamal) [reply ui/.eslintrc.js:63]: done :+1:

**Comment by leecalcote** [comment]
@Abhishek-kumar09 is this what you're looking for?

File: ui/.eslintrc.js (hunk 2 above)
Comment (Abhishek-kumar09) [inline ui/.eslintrc.js:45]: don't need to make changes here. Please revert.

File: ui/.eslintrc.js [hunk 3]
```diff
@@ -65,15 +61,14 @@ module.exports = {
     ],
     "react/react-in-jsx-scope" : "off",
     "no-undef" : "error",
-    "react/prop-types" : 0,
```
Comment (Abhishek-kumar09) [inline ui/.eslintrc.js:68]: same here

**Review by Abhishek-kumar09 — CHANGES_REQUESTED** [review]

File: ui/.eslintrc.js (hunk 2 above)
Comment (legendarykamal) [reply ui/.eslintrc.js]: oh ok done, I notice their order just now and feel how different they are :+1:

**Comment by Abhishek-kumar09** [comment]
need to fix eslint errors here: https://github.com/meshery/meshery/runs/8267015723?check_suite_focus=true

**Comment by legendarykamal** [comment]
@Abhishek-kumar09 I will resolve 2 of them but other 2 occur as you had asked me to revert the changes done on duplicate keys . So what should I do with them . 
Reference ->  ui/.eslintrc.js file

**Comment by Abhishek-kumar09** [comment]
Okay okay, you must have told me that those were duplicates keys you had removed from eslintrc, yeah, sorry, please remove them from the eslintrc also.

**Comment by leecalcote** [comment]
@Devansu-Yadav you might have a look here and offer any comments.

**Comment by legendarykamal** [comment]
Wohoo :balloon:  all the eslint errors have passed :) @Abhishek-kumar09

**Comment by Devansu-Yadav** [comment]
> @Devansu-Yadav you might have a look here and offer any comments.
Changes look fine to me and it seems like @legendarykamal was able to make all the ESlint checks run successfully after all 😄

**Review by Abhishek-kumar09 — APPROVED** [review]
Great, great, great. Thanks @legendarykamal

**Pull request merged**